src/main/java/com/sngular/kloadgen/sampler/schemaregistry/impl/ApicurioSchemaRegistry.java

```diff
@@ -0,0 +1,174 @@
+    try {
+      final SearchedArtifact searchedArtifact = getLastestSearchedArtifact(subjectName);
+      return this.schemaRegistryClient.getArtifactMetaData(SchemaRegistryKeyHelper.SCHEMA_REGISTRY_DEFAULT_GROUP, searchedArtifact.getId());
+    } catch (RestClientException e) {
+      throw new KLoadGenException(e.getMessage());
+    }
+  }
+
+  @Override
+  public ApicurioParsedSchema getSchemaBySubject(String subjectName) {
+    final ApicurioParsedSchema schema = new ApicurioParsedSchema();
+    try {
+      List<SearchedArtifact> artifacts = this.schemaRegistryClient.searchArtifacts(null, subjectName, null,
+                                                                                   null, null, null, null, null, null).getArtifacts();
+      if (artifacts.isEmpty()) {
+        throw new KLoadGenException(String.format("Schema %s not found", subjectName));
+      } else {
+        SearchedArtifact searchedArtifact = artifacts.get(0);
+        InputStream inputStream = this.schemaRegistryClient.getLatestArtifact(searchedArtifact.getGroupId(), searchedArtifact.getId());
+        String result = IOUtils.toString(inputStream, String.valueOf(StandardCharsets.UTF_8));
+
+        String searchedArtifactType = searchedArtifact.getType().toString();
+        if (SchemaTypeEnum.AVRO.name().equalsIgnoreCase(searchedArtifactType)) {
+          SchemaParser parserAvro = new AvroSchemaParser(null);
+          schema.setSchema(parserAvro.parseSchema(result.getBytes(StandardCharsets.UTF_8), new HashMap<>()));
+        } else if (SchemaTypeEnum.JSON.name().equalsIgnoreCase(searchedArtifactType)) {
+          SchemaParser parserJson = new JsonSchemaParser();
+          schema.setSchema(parserJson.parseSchema(result.getBytes(StandardCharsets.UTF_8), new HashMap<>()));
+        } else if (SchemaTypeEnum.PROTOBUF.name().equalsIgnoreCase(searchedArtifactType)) {
+          SchemaParser parserProtobuf = new ProtobufSchemaParser();
+          schema.setSchema(parserProtobuf.parseSchema(result.getBytes(StandardCharsets.UTF_8), new HashMap<>()));
+        } else {
+          throw new KLoadGenException(String.format("Schema type not supported %s", searchedArtifactType));
+        }
+        schema.setType(searchedArtifactType);
+        return schema;
+      }
+    } catch (IOException e) {
+      throw new KLoadGenException(e);
+    }
+  }
+
+  public Object getSchemaBySubjectAndId(String subjectName, Object metadata) {
+    Object schema = null;
+    final ArtifactMetaData schemaMetadata = (ArtifactMetaData) metadata;
+
+    try {
+      InputStream inputStream = this.schemaRegistryClient.getContentByGlobalId(schemaMetadata.getGlobalId());
+      String result = IOUtils.toString(inputStream, String.valueOf(StandardCharsets.UTF_8));
+
+      String searchedArtifactType = schemaMetadata.getType().toString();
+      if (SchemaTypeEnum.AVRO.name().equalsIgnoreCase(searchedArtifactType)) {
```

Comment:
Another way (if you don't mind to use `case`):
```
SchemaParser parser;
switch (SchemaTypeEnum.valueOf(searchedArtifactType.toUpperCase())) {
  case AVRO:
    parser = new AvroSchemaParser(null);
    break;
    ...
}
schema = parser.parseSchema(result.getBytes(StandardCharsets.UTF_8), new HashMap<>());
```

Reply:
I don't mind not using any design pattern. In that class there are several candidate block statements for refactor